src/operator/tensor/elemwise_unary_op_basic.cc

```diff
@@ -399,6 +399,71 @@ NNVM_REGISTER_OP(reshape_like)
 .add_argument("lhs", "NDArray-or-Symbol", "First input.")
 .add_argument("rhs", "NDArray-or-Symbol", "Second input.");
 
+NNVM_REGISTER_OP(shape_nd)
+.describe(R"code(Returns a 1D int64 array containing the shape of data.
+
+Example::
+
+  shape_nd([[1,2,3,4], [5,6,7,8]]) = [2,4]
+
+)code" ADD_FILELINE)
+.set_num_inputs(1)
+.set_num_outputs(1)
+.set_attr<nnvm::FIgnoreInputs>("FIgnoreInputs",
+    [](const NodeAttrs& attrs) { return std::vector<uint32_t>(1, 1); })
+.set_attr<FCompute>("FCompute<cpu>", ShapeCompute<cpu>)
+.set_attr<nnvm::FInferShape>("FInferShape",
+   [](const nnvm::NodeAttrs& attrs,
+      std::vector<TShape> *in_attrs,
+      std::vector<TShape> *out_attrs) {
+       CHECK_EQ(in_attrs->size(), 1U);
+       CHECK_EQ(out_attrs->size(), 1U);
+       TShape target_shape(1);
+       target_shape[0] = in_attrs->at(0).ndim();
+       SHAPE_ASSIGN_CHECK(*out_attrs, 0, target_shape);
+       return !shape_is_none(out_attrs->at(0));
+     })
+.set_attr<nnvm::FInferType>("FInferType",
```

Comment:
not registering FGradient ?

Reply:
Shape operator does not have a differential. Check the conversation here - https://github.com/apache/incubator-mxnet/issues/10789#issuecomment-388274133